File: app/src/main/java/com/github/kornilovmikhail/homework/MusicPlayerActivity.java
package com.github.kornilovmikhail.homework;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.IBinder;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

public class MusicPlayerActivity extends AppCompatActivity implements View.OnClickListener, ServiceConnection {

    private ImageButton prevButton;
    private ImageButton pauseButton;
    private ImageButton nextButton;
    private TextView tvNameTrack;
    private AudioService audioService;
    private List<Track> tracks;
    private boolean isPlaying;
    private int position;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTheme(StyleChanger.getStyle(getApplicationContext()));
        setContentView(R.layout.activity_player);
        tracks = new TracksRepository(this).getTracks();
        Intent intent = getIntent();
        position = intent.getIntExtra(ListTrackActivity.INTENT_CODE_POSITION, 0);
        tvNameTrack = findViewById(R.id.tv_track_name_player);
        tvNameTrack.setText(tracks.get(position).getName());
        prevButton = findViewById(R.id.btn_previous_track);
        prevButton.setOnClickListener(this);
        pauseButton = findViewById(R.id.btn_pause_play);
        pauseButton.setOnClickListener(this);
        nextButton = findViewById(R.id.btn_next_track);
        nextButton.setOnClickListener(this);
        Intent intentService = new Intent(this, AudioService.class);
        startService(intentService);
        bindService(intentService, this, Context.BIND_AUTO_CREATE);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_previous_track:
                getPreviousTrack();
                break;
            case R.id.btn_pause_play:
                pausePlayTrack(view);
                break;
            case R.id.btn_next_track:
                getNextTrack();
                break;
        }
    }

    private void getPreviousTrack() {
        audioService.stopTrack();
        if (position == 0) {
            position = tracks.size() - 1;
        } else {
            position--;
        }
        tvNameTrack.setText(tracks.get(position).getName());
        audioService.updateState(tracks, position);
        if (isPlaying) {
            audioService.pausePlayTrack();
        }
    }

    private void pausePlayTrack(View view) {
        ImageView iv = (ImageView) view;
        if (!isPlaying) {
            if (view != null) {
                iv.setImageResource(R.drawable.ic_round_pause_black_48dp);
            }
            audioService.pausePlayTrack();
            isPlaying = true;
        } else {
            if (view != null) {
                iv.setImageResource(R.drawable.ic_round_play_arrow_black_48dp);
            }
            audioService.pausePlayTrack();
            isPlaying = false;
        }
    }

    private void getNextTrack() {
        audioService.stopTrack();
        if (position == tracks.size() - 1) {
            position = 0;
        } else {
            position++;
        }
        tvNameTrack.setText(tracks.get(position).getName());
        audioService.updateState(tracks, position);
        if (isPlaying) {
            audioService.pausePlayTrack();
        }
    }

    @Override
    public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
        AudioService.TrackBinder audioServiceBinder = (AudioService.TrackBinder) iBinder;
        audioService = audioServiceBinder.getService();
        audioService.updateState(tracks, position);
    }

    @Override
    public void onServiceDisconnected(ComponentName componentName) {
        unbindService(this);
    }
}

File: app/src/main/java/com/github/kornilovmikhail/homework/MusicPlayerActivity.kt
package com.github.kornilovmikhail.homework

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.os.IBinder
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import kotlinx.android.synthetic.main.activity_player.*

class MusicPlayerActivity : AppCompatActivity(), View.OnClickListener, ServiceConnection {

    private var audioService: AudioService? = null
    private lateinit var tracks: List<Track>
    private var isPlaying: Boolean = false
    private var position: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(StyleChanger.getStyle(applicationContext))
        setContentView(R.layout.activity_player)
        tracks = TracksRepository(this).getTracks() as List<Track>
        val intent = intent
        position = intent.getIntExtra(ListTrackActivity.INTENT_CODE_POSITION, 0)

        tv_track_name_player.text = tracks[position].name
        btn_previous_track.setOnClickListener(this)
        btn_pause_play.setOnClickListener(this)
        btn_next_track.setOnClickListener(this)
        val intentService = Intent(this, AudioService::class.java)
        startService(intentService)
        bindService(intentService, this, Context.BIND_AUTO_CREATE)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.btn_previous_track -> getPreviousTrack()
            R.id.btn_pause_play -> pausePlayTrack(view)
            R.id.btn_next_track -> getNextTrack()
        }
    }

    private fun getPreviousTrack() {
        audioService?.stopTrack()
        if (position == 0) {
            position = tracks.size - 1
        } else {
            position--
        }
        tv_track_name_player.text = tracks[position].name
        audioService?.updateState(tracks, position)
        if (isPlaying) {
            audioService?.pausePlayTrack()
        }
    }

    private fun pausePlayTrack(view: View?) {
        val iv = view as ImageView?
        isPlaying = if (!isPlaying) {
            if (view != null) {
                iv?.setImageResource(R.drawable.ic_round_pause_black_48dp)
            }
            audioService?.pausePlayTrack()
            true
        } else {
            if (view != null) {
                iv?.setImageResource(R.drawable.ic_round_play_arrow_black_48dp)
            }
            audioService?.pausePlayTrack()
            false
        }
    }

    private fun getNextTrack() {
        audioService?.stopTrack()
        if (position == tracks.size - 1) {
            position = 0
        } else {
            position++
        }
        tv_track_name_player.text = tracks[position].name
        audioService?.updateState(tracks, position)
        if (isPlaying) {
            audioService?.pausePlayTrack()
        }
    }

    override fun onServiceConnected(componentName: ComponentName?, iBinder: IBinder?) {
        val audioServiceBinder = iBinder as AudioService.TrackBinder
        audioService = audioServiceBinder.service
        audioService?.updateState(tracks, position)
    }


    override fun onServiceDisconnected(componentName: ComponentName) {
        unbindService(this)
    }
}
